tests/data_types/static_assert_tests/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+
+function(add_static_assert_test file error_msg)
+
+add_executable(${file}-build-fail-test
+    ${file}.cpp
+    ../../main.cpp
+)
+target_link_libraries(${file}-build-fail-test
+        PUBLIC
+        GTest::gtest
+        GTest::gmock
```

Comment:
Can we get rid off google test for compilation-only tests ?

Reply:
Apparently yes : f397e05f

Reply:
I mean, `add_static_assert_test` should not even need googletest as it does not use any runtime `ASSERT`/`EXPECT` right ? I think if we also remove `GTest::gmock` (that actually brings transitively `GTest::gtest`) we should be fine. The file `main.cpp` will complain but we don't need it either.

Reply:
I think you need at least one for the `#include <gtest/gtest.h>` to be correctly interpreted.

Reply:
What about writing the tests like this:
```cpp
using Tensor2D_A = Tensor<int, VectorIndexSet<R, Theta>, VectorIndexSet<R_cov, Theta_cov>>;
using Tensor2D_B = Tensor<int, VectorIndexSet<X, Y>, VectorIndexSet<R_cov, Theta_cov>>;
using Tensor2D_C = Tensor<int, VectorIndexSet<R, Theta>, VectorIndexSet<R_cov, Theta_cov>>;

Tensor2D_C compile_tensor_test_mul(Tensor2D_A& A, Tensor2D_B& B)
{
    ddcHelper::get<R, R>(A) = 1;
    ddcHelper::get<R, Theta>(A) = 0;
    ddcHelper::get<Theta, R>(A) = 2;
    ddcHelper::get<Theta, Theta>(A) = 4;
    ddcHelper::get<X, R_cov>(B) = 6;
    ddcHelper::get<X, Theta_cov>(B) = 8;
    ddcHelper::get<Y, R_cov>(B) = 4;
    ddcHelper::get<Y, Theta_cov>(B) = 5;
    return tensor_mul(index<'i', 'j'>(A), index<'j', 'k'>(B));
}
```

Reply:
Ah ok. Why not, maybe it will make the compilation faster :+1:

Reply:
Done : 100615b11fab83dfea2bc81e2f2d37fe0d1485e5

Reply:
We should think of making this snippet of cmake code available for other projects. I don't know if it is robust enough to capture correctly the message in the `static_assert` with other compilers than the ones in the CI ?

Reply:
I found the instructions on the internet so I'm not sure it's mine to share. I hope it is robust enough as it just checks that the static assert message is raised. That should always be the case unless the assertion is broken I think?